python/pyspark/sql/dataframe.py

```diff
@@ -1311,6 +1328,23 @@ def fill(self, value, subset=None):
     fill.__doc__ = DataFrame.fillna.__doc__
 
 
+class DataFrameStatFunctions(object):
+    """Functionality for statistic functions with :class:`DataFrame`.
+    """
+
+    def __init__(self, df):
+        self.df = df
+
+    def cov(self, col1, col2):
+        if not isinstance(col1, str):
+            raise ValueError("col1 should be a string.")
+        if not isinstance(col2, str):
+            raise ValueError("col2 should be a string.")
+        return self.df._jdf.stat().cov(col1, col2)
+
+    cov.__doc__ = DataFrame.cov.__doc__
```

Comment:
if you do the doc assignment here, then you should move the implementation to dataframe also ... or just put the doc assignment there. don't put the doc in one place and implementation in another